packages/framework/memo-cache/src/getMemoCache.ts

```diff
@@ -0,0 +1,20 @@
+import { CacheEntry, getCacheEntry } from './getCacheEntry';
+
+export type ValueFactory<T> = () => T;
+export type GetMemoValue<T, TGet = any> = (factory: T | ValueFactory<T>, keys: any[]) => [T, GetMemoValue<TGet>];
+
+const _baseEntry: CacheEntry<any> = {};
+
+function getMemoValueWorker<T, TGet = any>(entry: CacheEntry<any>, factory: T | ValueFactory<T>, keys: any[]): [T, GetMemoValue<TGet>] {
+  const foundEntry = getCacheEntry(entry, keys);
+  // check the key being set, not the value to disambiguate an undefined factory result/value from never having run the factory
+  if (!foundEntry.hasOwnProperty('value') && factory !== undefined && factory !== null) {
+    foundEntry.value = typeof factory === 'function' ? (factory as ValueFactory<T>)() : factory;
+  }
+  return [foundEntry.value, (fact: TGet | ValueFactory<TGet>, args: any[]) => getMemoValueWorker<TGet>(foundEntry, fact, args)];
```

Comment:
What's the desired behavior when `factory = null`? I see the tests check for value being `undefined`, but it's not particularly crazy to say that `null` is a valid value and that if I pass `null` in as my factory I expect to get `null` back and not `undefined`, right? That said, it is a strange case.

Reply:
The reason for it is to just make it a no-op in the case where you just want to retrieve the sub-cache.  Originally it didn't support void functions (really only useful for the memoize helper) and when I added void function support and memoize support (one of the last changes) I didn't update it. Frankly I don't think this is a real pattern and it should probably just be the hasOwnProperty check.

Reply:
Now passing null returns null and undefined returns undefined correctly with the change.